src/listen/websocket.rs

```diff
@@ -267,35 +270,34 @@ where
                     pairs.append_pair(&key, &value);
                 }
             }
-            if let Some(encoding) = &self.encoding {
+            if let Some(encoding) = &self.builder.encoding {
                 pairs.append_pair("encoding", encoding.as_str());
             }
-            if let Some(sample_rate) = self.sample_rate {
+            if let Some(sample_rate) = self.builder.sample_rate {
                 pairs.append_pair("sample_rate", &sample_rate.to_string());
             }
-            if let Some(channels) = self.channels {
+            if let Some(channels) = self.builder.channels {
                 pairs.append_pair("channels", &channels.to_string());
             }
-            if let Some(endpointing) = self.endpointing {
+            if let Some(endpointing) = self.builder.endpointing {
                 pairs.append_pair("endpointing", &endpointing.to_str());
             }
-            if let Some(utterance_end_ms) = self.utterance_end_ms {
+            if let Some(utterance_end_ms) = self.builder.utterance_end_ms {
                 pairs.append_pair("utterance_end_ms", &utterance_end_ms.to_string());
             }
-            if let Some(interim_results) = self.interim_results {
+            if let Some(interim_results) = self.builder.interim_results {
                 pairs.append_pair("interim_results", &interim_results.to_string());
             }
-            if let Some(no_delay) = self.no_delay {
+            if let Some(no_delay) = self.builder.no_delay {
                 pairs.append_pair("no_delay", &no_delay.to_string());
             }
-            if let Some(vad_events) = self.vad_events {
+            if let Some(vad_events) = self.builder.vad_events {
                 pairs.append_pair("vad_events", &vad_events.to_string());
             }
         }
 
         let mut source = self
-            .source
-            .ok_or(DeepgramError::NoSource)?
```

Comment:
With the new typing, this error is unnecessary.  You can't call `.start()` until you've attached a source.  (I'll remove that error type in just a moment.